### PR TITLE
book: Include ashpd only on linux so that listings run on macos

### PR DESCRIPTION
**Pull request by daxhuiberts**
This will make `ashpd` only a dependency on linux.

An alternative would be something like `cfg(all(unix, not(target_os = "macos")))` to still include it for all other unix family.

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
The code using it should also be behind a similar guard

**Comment by daxhuiberts** [comment]
> The code using it should also be behind a similar guard

I looked into disabling the `main_event_loop_7` bin, but there currently is no simple way to achieve that (see https://github.com/rust-lang/cargo/issues/9208)
Not sure what the appropriate approach would be in this case.

**Comment by daxhuiberts** [comment]
I made a change to `main_event_loop_7` with a possible solution, but this approach is probably not desirable.

File: book/listings/main_event_loop/7/main.rs [hunk 1]
```diff
@@ -65,3 +49,30 @@ fn build_ui(app: &Application) {
...
+        println!("User name: {}", response.name());
+    } else {
+        println!("Could not access user information.")
+    }
+}
+
+#[cfg(not(target_os = "linux"))]
+async fn fetch_user_information(_button: Button) {
+    println!("fetching user information not available outside target_os = \"linux\"");
+}
```
Comment (bilelmoussaoui) [inline book/listings/main_event_loop/7/main.rs:78]: Missing newline at the end of the file

**Comment by bilelmoussaoui** [comment]
Also, please fix the formatting, otherwise the suggested solution is the best we can do so it is fine.

**Comment by daxhuiberts** [comment]
Thanks for the feedback.
Fixed the newline and formatting.

**Comment by bilelmoussaoui** [comment]
Thanks!

**Pull request merged**